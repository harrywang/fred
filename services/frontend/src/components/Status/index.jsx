import React, { useState } from 'react'
import axios from 'axios'
import styles from './index.module.scss'
import { Redirect } from 'react-router-dom'
import { useEffect } from 'react'

const Status = props => {
  const { isAuthenticated, accessToken } = props

  const [state, setState] = useState({ email: '', username: '' })

  const getUserStatus = () => {
    const options = {
      url: `${process.env.REACT_APP_BACKEND_SERVICE_URL}/auth/status`,
      method: 'get',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${accessToken}`,
      },
    }
    return axios(options)
      .then(res => {
        setState({
          email: res.data.email,
          username: res.data.username,
        })
      })
      .catch(error => {
        console.log(error)
      })
  }

  useEffect(() => {
    getUserStatus()
  }, [])

  if (!isAuthenticated()) {
    return <Redirect to="/login" />
  }

  return isAuthenticated() ? (
    <div className={styles.status_wrap}>
      <div className={styles.status_title}>User Status</div>
      <div className={styles.status_content}>
        <strong>Email: </strong>
        <span data-testid="user-email">{state.email}</span>
        <br />
        <strong>Username: </strong>
        <span data-testid="user-username">{state.username}</span>
      </div>
    </div>
  ) : (
    <Redirect to="/login" />
  )
}

export default Status
