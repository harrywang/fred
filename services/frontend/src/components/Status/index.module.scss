.status_wrap {
    padding: 0 48px;
    max-width: 1120px;
    margin: 0 auto;
}

.status_title {
    margin-top: 6rem;
    margin-bottom: 3rem;
    font-family: 'Montserrat', sans-serif !important;
    color: #444f60;
    font-size: 2rem;
    font-weight: 600;
    line-height: 1.125;
    text-align: center;
  }

.status_content {
    width: 50%;
    margin: 0 auto;
}