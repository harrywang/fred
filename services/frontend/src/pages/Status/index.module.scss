.status_wrap {
    padding: 0 48px;
    max-width: 1120px;
    margin: 0 auto;
}

.status_title {
    padding: 6rem 0;
    font-family: 'Montserrat', sans-serif !important;
    color: #444f60;
    margin-bottom: 1.5rem;
    font-size: 2rem;
    font-weight: 600;
    line-height: 1.125;
    text-align: center;
  }

.status_content {
    width: 50%;
    margin: 0 auto;
}