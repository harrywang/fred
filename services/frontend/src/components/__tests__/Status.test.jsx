import React from 'react'
import { cleanup, wait } from '@testing-library/react'
import axios from 'axios'

import Status from '../Status'

afterEach(cleanup)

jest.mock('axios')

axios.mockImplementation(() =>
  Promise.resolve({
    data: { email: 'test@test.com', username: 'test' },
  }),
)

const props = {
  isAuthenticated: () => {
    return true
  },
}

test('renders properly when authenticated', async () => {
  const { container, findByTestId } = renderWithRouter(<Status {...props} />)
  await wait(() => {
    expect(axios).toHaveBeenCalledTimes(1)
  })
  expect((await findByTestId('user-email')).innerHTML).toBe('test@test.com')
  expect((await findByTestId('user-username')).innerHTML).toBe('test')
})

test('renders', async () => {
  const { asFragment } = renderWithRouter(<Status {...props} />)
  await wait(() => {
    expect(axios).toHaveBeenCalled()
  })
  expect(asFragment()).toMatchSnapshot()
})
