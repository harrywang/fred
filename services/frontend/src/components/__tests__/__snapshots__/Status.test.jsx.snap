// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="status_wrap"
  >
    <div
      class="status_title"
    >
      User Status
    </div>
    <div
      class="status_content"
    >
      <strong>
        Email: 
      </strong>
      <span
        data-testid="user-email"
      >
        test@test.com
      </span>
      <br />
      <strong>
        Username: 
      </strong>
      <span
        data-testid="user-username"
      >
        test
      </span>
    </div>
  </div>
</DocumentFragment>
`;
